djangocms_googlemap/models.py

```diff
@@ -1,90 +1,288 @@
-        'address', 'zipcode', 'width', 'height']
+MAP_TYPES = ['ROADMAP', 'SATELLITE', 'HYBRID', 'TERRAIN']
+MAP_TYPE_CHOICES = [(map_type, map_type) for map_type in MAP_TYPES]
+TRAVEL_MODES = ['DRIVING', 'BICYCLING', 'TRANSIT', 'WALKING']
+TRAVEL_MODE_CHOICES = [(travel_mode, travel_mode) for travel_mode in TRAVEL_MODES]
+ZOOM_LEVEL_CHOICES = [(c, str(c)) for c in range(22)] # from 0 to 21
 
-    title = models.CharField(_("map title"), max_length=100, blank=True,
-                             null=True)
+CSS_WIDTH_RE = re.compile(r'^\d+(?:px|%)$')
+CSS_HEIGHT_RE = re.compile(r'^\d+px$')
 
-    address = models.CharField(_("address"), max_length=150)
-    zipcode = models.CharField(_("zip code"), max_length=30)
-    city = models.CharField(_("city"), max_length=100)
 
-    content = models.CharField(
-        _("additional content"), max_length=255, blank=True,
-        help_text=_('Displayed under address in the bubble.'))
+# Add additional choices through the ``settings.py``.
+def get_templates():
+    choices = [
+        ('default', _('Default')),
+    ]
+    choices += getattr(
+        settings,
+        'DJANGOCMS_GOOGLEMAP_TEMPLATES',
+        [],
+    )
+    return choices
 
-    style = models.TextField(_("custom map style"), blank=True,
-        help_text=_('Provide a valid JSON configuration (escaped). See '
-            'developers.google.com/maps/documentation/javascript/styling'))
 
-    ZOOM_LEVELS = map(lambda c: (c, str(c)), range(22))
+@python_2_unicode_compatible
+class GoogleMap(CMSPlugin):
+    """
+    Renders the Google Maps wrapper
+    """
+    template = models.CharField(
+        verbose_name=_('Template'),
+        choices=get_templates(),
+        default=get_templates()[0][0],
+        max_length=255,
+    )
+    title = models.CharField(
+        verbose_name=_('Map title'),
+        max_length=255,
+        blank=True,
+    )
+    width = models.CharField(
+        verbose_name=_('Width'),
+        max_length=6,
+        default='100%',
+        help_text=_('Width of the map, including the CSS length units (e.g. "100%", "400px" or "400rem").'),
+    )
+    height = models.CharField(
+        verbose_name=_('Height'),
+        max_length=6,
+        default='400px',
+        help_text=_('Height of the map, including the CSS length units (e.g. "400px" or "400rem").'),
+    )
+    style = models.TextField(
+        verbose_name=_('Map styling'),
+        blank=True,
+        help_text=_('Provide a valid (escaped) JSON configuration. See '
+            'http://developers.google.com/maps/documentation/javascript/styling'),
+    )
+    lat = models.FloatField(
+        verbose_name=_('Latitude (lat)'),
+        null=True,
+        blank=True,
+        help_text=_('Default Geographical latitute (e.g. "46.947973").'),
+    )
+    lng = models.FloatField(
+        verbose_name=_('Longitude (lng)'),
+        null=True,
+        blank=True,
+        help_text=_('Default Geographical longitude (e.g. "7.447446").'),
```

Comment:
``Geographical longitude in degrees (e.g. "7.447446").`` for consistency with the help_text in GoogleMapMarker.